frontend/fingertips-frontend/components/viewPlots/TwoOrMoreIndicatorsAreasViewPlots/index.tsx

```diff
@@ -99,6 +99,38 @@ export function mapToSpineChartTableProps(
   return { rowData: tableData };
 }
 
+export function extractHeatmapIndicatorData(
+  indicatorData: IndicatorWithHealthDataForArea,
+  metadata?: IndicatorDocument
+): HeatmapIndicatorData {
+  const getIndicatorId = (): string => {
+    if (metadata) {
+      return metadata.indicatorID;
+    }
+
+    return indicatorData.indicatorId !== undefined
+      ? indicatorData.indicatorId?.toString()
+      : 'undefined indicator id';
+  };
+
+  const getIndicatorName = (): string => {
+    if (metadata) {
+      return metadata.indicatorName;
+    }
+
+    return indicatorData.name ?? 'undefined indicator name';
+  };
+
+  return {
+    indicatorId: getIndicatorId(),
+    indicatorName: getIndicatorName(),
```

Comment:
not sure why we have the little helper functions here and just do
```
    indicatorId: metadata?.indicatorID,
    indicatorName: metadata?.indicatorName,
```
Also why are we setting these values to string `undefined indicator id` if it doesn't exist. Wouldn't it being undefined work better as a type.

If these values don't exist in the metadata would showing those string values be ok?